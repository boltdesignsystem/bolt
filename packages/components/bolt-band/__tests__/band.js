import {
  render,
  renderString,
  stopServer,
} from '../../../testing/testing-helpers';
import schema from '../band.schema';
const { size, theme, tag } = schema.properties;

const timeout = 120000;

describe('<bolt-band> Component', () => {
  let page;

  beforeEach(async () => {
    await page.evaluate(() => {
      document.body.innerHTML = '';
    });
  }, timeout);

  beforeAll(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
    });
  }, timeout);

  afterAll(async () => {
    await stopServer();
    await page.close();
  }, 100);

  // Basic Usage
  test('Basic usage', async () => {
    const results = await render('@bolt-components-band/band.twig', {
      content: 'This is a band.',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  size.enum.forEach(async sizeChoice => {
    test(`Size of the band: ${sizeChoice}`, async () => {
      const results = await render('@bolt-components-band/band.twig', {
        size: sizeChoice,
        content: 'This is a band.',
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  theme.enum.forEach(async themeChoice => {
    test(`Theme of the band: ${themeChoice}`, async () => {
      const results = await render('@bolt-components-band/band.twig', {
        theme: themeChoice,
        content: 'This is a band.',
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  tag.enum.forEach(async tagChoice => {
    test(`Semantic tag usage: ${tagChoice}`, async () => {
      const results = await render('@bolt-components-band/band.twig', {
        tag: tagChoice,
        content: 'This is a band.',
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  test('Full bleed usage', async function() {
    const renderedHTML = await renderString(`
      {% include "@bolt-components-band/band.twig" with {
        theme: "dark",
        full_bleed: false,
        content: "This band is not full bleed.",
      } only %}
      {% include "@bolt-components-band/band.twig" with {
        theme: "xdark",
        full_bleed: true,
        content: "This band is full bleed.",
      } only %}
    `);

    expect(renderedHTML).toMatchSnapshot();
  });

  test(`Pinned content usage`, async () => {
    const results = await render('@bolt-components-band/band.twig', {
      size: 'large',
      content: 'This is a band.',
      pinned_content: {
        upper: [
          {
            content: 'upper pinned 1',
          },
          {
            content: 'upper pinned 2',
          },
          {
            content: 'upper pinned 3',
          },
        ],
        lower: [
          {
            content: 'lower pinned 1',
          },
          {
            content: 'lower pinned 2',
          },
          {
            content: 'lower pinned 3',
          },
        ],
      },
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Nested bands usage', async function() {
    const renderedHTML = await renderString(`
      {% set parent_band_content %}
        {% include "@bolt-components-headline/headline.twig" with {
          size: "xxxlarge",
          text: "This Band Has 2 Bands Nested Inside",
        } only %}
        {% include "@bolt-components-band/band.twig" with {
          size: "xsmall",
          theme: "light",
          full_bleed: false,
          content: "Nested band 1",
        } only %}
        {% include "@bolt-components-band/band.twig" with {
          size: "large",
          theme: "xlight",
          full_bleed: false,
          content: "Nested band 2",
        } only %}
      {% endset %}
      {% include "@bolt-components-band/band.twig" with {
        theme: "xdark",
        content: parent_band_content,
      } only %}
    `);

    expect(renderedHTML).toMatchSnapshot();
  });
});
