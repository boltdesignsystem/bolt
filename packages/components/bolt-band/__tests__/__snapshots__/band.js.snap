// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-band> Component Basic usage 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Full bleed usage 1`] = `
Object {
  "html": <bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This band is not full bleed.
      </div>
    </div>
  </div>
</bolt-band>
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-xdark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This band is full bleed.
      </div>
    </div>
  </div>
</bolt-band>,
  "message": "",
  "ok": true,
}
`;

exports[`<bolt-band> Component Nested bands usage 1`] = `
Object {
  "html": <bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-xdark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        <h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxxlarge">
          This Band Has 2 Bands Nested Inside
        </h2>
        <bolt-band class="is-ready">
          <div class="c-bolt-band c-bolt-band--xsmall c-bolt-band--valign-center t-bolt-light">
            <div class="c-bolt-band__content">
              <div class="e-bolt-wrapper">
                Nested band 1
              </div>
            </div>
          </div>
        </bolt-band>
        <bolt-band class="is-ready">
          <div class="c-bolt-band c-bolt-band--large c-bolt-band--valign-center t-bolt-xlight">
            <div class="c-bolt-band__content">
              <div class="e-bolt-wrapper">
                Nested band 2
              </div>
            </div>
          </div>
        </bolt-band>
      </div>
    </div>
  </div>
</bolt-band>,
  "message": "",
  "ok": true,
}
`;

exports[`<bolt-band> Component Pinned content usage 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--large c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content c-bolt-band__content--pinned c-bolt-band__content--pinned-upper">
      <div class="e-bolt-wrapper">
        <div class="c-bolt-band__pinned-items">
          <div class="c-bolt-band__pinned-item">
            upper pinned 1
          </div>
          <div class="c-bolt-band__pinned-item">
            upper pinned 2
          </div>
          <div class="c-bolt-band__pinned-item">
            upper pinned 3
          </div>
        </div>
      </div>
    </div>
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
    <div class="c-bolt-band__content c-bolt-band__content--pinned c-bolt-band__content--pinned-lower">
      <div class="e-bolt-wrapper">
        <div class="c-bolt-band__pinned-items">
          <div class="c-bolt-band__pinned-item">
            lower pinned 1
          </div>
          <div class="c-bolt-band__pinned-item">
            lower pinned 2
          </div>
          <div class="c-bolt-band__pinned-item">
            lower pinned 3
          </div>
        </div>
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Semantic tag usage: article 1`] = `
<bolt-band class="is-ready">
  <article class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </article>
</bolt-band>
`;

exports[`<bolt-band> Component Semantic tag usage: div 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Semantic tag usage: figure 1`] = `
<bolt-band class="is-ready">
  <figure class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </figure>
</bolt-band>
`;

exports[`<bolt-band> Component Semantic tag usage: footer 1`] = `
<bolt-band class="is-ready">
  <footer class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </footer>
</bolt-band>
`;

exports[`<bolt-band> Component Semantic tag usage: header 1`] = `
<bolt-band class="is-ready">
  <header class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </header>
</bolt-band>
`;

exports[`<bolt-band> Component Semantic tag usage: nav 1`] = `
<bolt-band class="is-ready">
  <nav class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </nav>
</bolt-band>
`;

exports[`<bolt-band> Component Semantic tag usage: section 1`] = `
<bolt-band class="is-ready">
  <section class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </section>
</bolt-band>
`;

exports[`<bolt-band> Component Size of the band: large 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--large c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Size of the band: medium 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Size of the band: none 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Size of the band: small 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--small c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Size of the band: xsmall 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--xsmall c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Theme of the band: dark 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Theme of the band: light 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-light c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Theme of the band: none 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Theme of the band: xdark 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-xdark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Theme of the band: xlight 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-xlight c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;

exports[`<bolt-band> Component Theme of the band: xxdark 1`] = `
<bolt-band class="is-ready">
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--valign-center t-bolt-xxdark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <div class="e-bolt-wrapper">
        This is a band.
      </div>
    </div>
  </div>
</bolt-band>
`;
